Automattic/jetpack
# AI Assistant: Support backend prompt building when composing forms

Fixes #32613.

Depends on D119078-code.

## Proposed changes:

* Change messages type to turn the `content` field optional and to add the `context` property
* Start sending the `context` on form completion requests

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Apply D119078-code to your sandbox and sandbox the `public-api` domain
* Go to the block editor and open the browser development console
* Insert a Jetpack Form block
* Ask for some form on the AI Extension (for example, a "burguer ordering form")
* Notice the prompt structure on the request debug logs:
   * Before this change, the full prompt for the form creation would be present on the request
   * After this change, we are adding a `context` property with the necessary context to build the prompt on the server side; the original prompt is still being sent so we do a swift transition

<img width="800" alt="Screenshot 2023-08-21 at 16 43 32" src="https://github.com/Automattic/jetpack/assets/6760046/e58fd69c-c0ac-4370-940a-72bc306932cd">

* Make sure the form is generated correctly
* Make sure followup requests are handled properly as well
